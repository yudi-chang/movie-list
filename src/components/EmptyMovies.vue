<template>
  <div class="not-found-message">
    <img src="/images/not-found-logo.png" alt="Movie not found" />
    <h2 class="mt-28 text-3xl fw-bold">No movie is found</h2>
  </div>
</template>

<style scoped lang="scss">
.not-found-message {
  text-align: center;
  width: 100%;
  padding: 70px 20px 0 20px;

  img {
    width: 200px;
    max-width: 100%;
    margin: 0 auto;
  }
}
</style>
