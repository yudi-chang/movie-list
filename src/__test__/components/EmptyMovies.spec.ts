import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import EmptyMovies from '@/components/EmptyMovies.vue'

describe('EmptyMovies', () => {
  const defaultParams = {}

  const createWrapper = (params = defaultParams) => {
    return mount(EmptyMovies, params)
  }

  it('should renders correctly', async () => {
    // no logic in this component, so just use matchSnapshot
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  })
})
