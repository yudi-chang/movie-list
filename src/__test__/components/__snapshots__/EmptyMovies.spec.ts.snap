// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmptyMovies > should renders correctly 1`] = `
<div
  class="not-found-message"
  data-v-20e20707=""
>
  <img
    alt="Movie not found"
    data-v-20e20707=""
    src="/images/not-found-logo.png"
  />
  <h2
    class="mt-28 text-3xl fw-bold"
    data-v-20e20707=""
  >
    No movies found
  </h2>
</div>
`;
